girder/utility/s3_assetstore_adapter.py

```diff
@@ -468,6 +468,14 @@ def importData(self, parent, parentType, params, progress,
                     progress.update(message=obj['Prefix'])
 
                 name = obj['Prefix'].rstrip('/').rsplit('/', 1)[-1]
+                # If there is already an item with the folder's name, append
+                # '/'.  This is what S3 does internally, allowing a folder and
+                # file to have the same name once stripped of the right /.
```

Comment:
To be a bit pedantic, the internal reason is because objects in S3 are not actually hierarchical, they're a flat list a la materialized paths. The presentation of things in S3 as folders is a bit of an illusion; objects don't have the same name, they simply share a common prefix.

I think this is a fine accommodation of that behavior, but maybe we could word the comment more clearly.